install/nodejs

```diff
@@ -0,0 +1,244 @@
+        ewarn "The user will need to assume responsibility for proper nodejs installation."
+        return 2
+    else
+        edebug "npm detected as a symlink, continuing installation"
+    fi
+
+    # If the path check didn't fail, and the path isn't a symlink then we abort
+    if [[ "${nodejs_path}" != "failed" && ! -L "${nodejs_path}" ]]; then
+        eerror "node (nodejs) has been detected as a non-symlink (${nodejs_path}), stopping installation."
+        ewarn "The user will need to assume responsibility for proper nodejs installation."
+        return 3
+    else
+        edebug "nodejs detected as a symlink, continuing installation"
+    fi
+
+    return 0
+}
+
+opt_usage cleanup_node_symlinks <<'END'
+Cleans up symlinks created from a prior installation NOTE: This is meant to be run before installation!
+END
+cleanup_node_symlinks()
+{
+    $(opt_parse                                                            \
+        ":install_prefix p | The install prefix"
+    )
+
+    local symlink_paths=()
+    eprogress "Cleaning up existing symlinks"
+
+    # Assumption is that if /usr/local/node is a non symlink, it should be left alone
+    if [[ ! -L "${install_prefix}/node" ]]; then
+        die "${install_prefix}/node is already defined and not a symlink, failing"
+    fi
+
+    # Grab previous install's binaries so their symlinks can be properly cleaned up before linking new binaries
+    symlink_paths=(/usr/local/node/bin/*)
+    orig_install=$(ls -al ${install_prefix}/node | awk '{print $11}')
+
+    # For all the binaries in the nodejs_dir, delete corresponding symlinks
+    for spath in "${symlink_paths[@]}"; do
+        local check_file="${install_prefix}/bin/$(basename "${spath}")"
+
+        if [[ -e "${check_file}" && ! -L "${check_file}" ]]; then
+            ewarn "Previous node configuration found! ${check_file}"
+            die "${check_file} does not appear to be a symlink, failing!"
+        fi
+        edebug "Removing previous symlink to ${check_file}"
+        rm -f "${check_file}"
+    done
+
+    # Remove existing dir symlink as it will be replaced by current lts
+    edebug "Removing previous node symlink: ${install_prefix}/node -> ${orig_install}"
+    rm -f "${install_prefix}/node"
+
+    eprogress_kill
+}
+
+
+opt_usage retrieve_sha <<'END'
+Retrieves nodejs release sha for the lts codename
+END
+retrieve_sha()
+{
+    $(opt_parse                                                    \
+        ":lts_codename     c | The codename of the lts to install" \
+        ":sha_output_path  o | The place to store the sha"         \
+    )
+
+    einfo "Retrieving SHASUM256.txt for codename ${lts_codename}"
+    local sha_url="https://nodejs.org/download/release/latest-${lts_codename}/SHASUMS256.txt"
+    edebug "Using url: ${sha_url}"
+
+    # Download SHA256 to determine version
+    eretry efetch --style=einfo "${sha_url}" "${sha_output_path}" | edebug
```

Comment:
efetch writes to both stdout and stderr, so you might want to use `|&` here to send both to edebug